server/client.go

```diff
@@ -879,8 +896,10 @@ func (c *client) processMsg(msg []byte) {
 			qsubs := r.qsubs[i]
 			index := c.cache.prand.Intn(len(qsubs))
 			sub := qsubs[index]
-			mh := c.msgHeader(msgh[:si], sub)
-			c.deliverMsg(sub, mh, msg)
+			if sub != nil {
```

Comment:
If `len(qsubs)` is 0, `prand.Intn()` would panic. If `len(qsubs) > 0`, then index should be `[0, n)`. How then `qsubs[index]` be `nil`? And if `sub` can really be `nil`, should the test be done just after `sub := qsubs[index]` and skip all the rest?